workspaces/meewee-time-registration/plugins/plugin-meewee-time-registration/meewee-time-registration/src/context/AuthContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { createContext, useReducer, useEffect } from 'react';
+import { AuthAction, AuthContextProps, AuthState } from '../types/authContext';
+
+const initialState: AuthState = {
+  isAuthenticated: false,
+  token: null,
+  user: null,
+};
+
+const AuthContext = createContext<AuthContextProps | undefined>(undefined);
+
+const authReducer = (state: AuthState, action: AuthAction): AuthState => {
+  switch (action.type) {
+    case 'LOGIN':
+      return {
+        ...state,
+        isAuthenticated: true,
+        user: action.payload.userName,
+        token: action.payload.token,
+      };
+    case 'LOGOUT':
+      return { ...state, isAuthenticated: false, user: null, token: null };
+    default:
+      return state;
+  }
+};
+
+const AuthProvider: React.FC = ({ children }) => {
```

Comment:
Sorry, want to avoid this again, no `React.FC`.

Reply:
Done👍